### PR TITLE
Feat/bit domain ownership pending

### PR DESCRIPTION
**Pull request by underdarknl**
### Changes

Some Registrars require periodic ownership verification. If this is not done in time, the dns servers are replaced by a placeholder DNS-Server showing a verification page. The domain does not function normally during this time.

### Issue link

Closes ...

### QA notes

Add one of the listed dns servers to the domain.

    'NS1.REGISTRANT-VERIFICATION.ISPAPI.NET',
    'NS2.REGISTRANT-VERIFICATION.ISPAPI.NET',
    'NS3.REGISTRANT-VERIFICATION.ISPAPI.NET'

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/bits/domain_owner_verification/domain_owner_verification.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    https://support.dnsimple.com/articles/icann-domain-validation/
+    """
+    if DNSNSRecord.name_server_hostname in INDICATORS:
+        for hostname in additional_oois:
+            finding_type = KATFindingType(id="KAT-DOMAIN-OWNERSHIP-PENDING")
+            yield finding_type
+            yield Finding(
+                finding_type=finding_type.reference,
+                ooi=hostname.reference,
+                description="This domain requires Ownership verification and is currently pending.",
```
Comment (ammar92) [inline octopoes/bits/domain_owner_verification/domain_owner_verification.py:31]:
```suggestion
                description="This domain requires ownership verification and is currently pending.",
```

File: octopoes/bits/domain_owner_verification/domain_owner_verification.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "NS2.REGISTRANT-VERIFICATION.ISPAPI.NET",
+    "NS3.REGISTRANT-VERIFICATION.ISPAPI.NET",
+]
+
+
+def run(nameserver: DNSNSRecord, additional_oois: list[Hostname], config: dict[str, str]) -> Iterator[OOI]:
+    """Checks to see if a domain has a specific set of dns servers which would indicate domain registrant verification.
+    https://support.dnsimple.com/articles/icann-domain-validation/
+    """
+    if DNSNSRecord.name_server_hostname in INDICATORS:
```
Comment (ammar92) [inline octopoes/bits/domain_owner_verification/domain_owner_verification.py:24]: Do we have to do some sanitization here (for casing in the text/ name) before searching, or can we assume the `name_server_hostname` is always in uppercase?

**Review by ammar92 — COMMENTED** [review]
Looks good to me 👍 Just one tiny question and a suggestion

File: octopoes/bits/domain_owner_verification/domain_owner_verification.py (hunk 2 above)
Comment (underdarknl) [reply octopoes/bits/domain_owner_verification/domain_owner_verification.py]: we do now :)

**Review by dekkers — COMMENTED** [review]
The blocking is done by the registrar and is registrar specific as far as I can see. The finding links to the dnsimple support page, but that seems to be only valid for dnsimple customers. For a different registers it will be different, see for example https://support.hostinger.com/en/articles/1583442-how-to-fix-a-domain-suspended-due-to-pending-icann-verification

The name-services.com DNS servers are also the general namservers of enom so they can't be used as indicator that the domain is suspended because of missing verification: https://support.enom.com/support/solutions/articles/201000065375-nameservers-ns-#h_01FR8WZG9C6Y5JJCJ4XMN3VEE5

File: boefjes/boefjes/plugins/kat_kat_finding_types/kat_finding_types.json [hunk 3]
```diff
@@ -486,5 +486,12 @@
     "source": "https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers",
     "impact": "Nonstandard headers may not be supported by all browsers and may not provide the security that is expected.",
     "recommendation": "Remove the nonstandard headers from the response."
+  },
+  "KAT-DOMAIN-OWNERSHIP-PENDING": {
+    "description": "The domain requires Ownership verification. An email hasbeen send to the registered owner by the registrar. The domain is currently offline.",
```
Comment (stephanie0x00) [inline boefjes/boefjes/plugins/kat_kat_finding_types/kat_finding_types.json:491]:
```suggestion
    "description": "The domain requires Ownership verification. An email has been send to the registered owner by the registrar. The domain is currently offline.",
```

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [ ] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [ ] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
All things unrelated to the PR seem to work. 

### What doesn't work:
See error message below. I manually added a DNSNSrecord and added one of the nameservers as hostname. 

### Bug or feature?:
n/a

```
octopoes_api_worker-1  | [2024-08-07 08:08:23,498: INFO/ForkPoolWorker-3] octopoes.tasks.tasks.recalculate_scan_profiles[ecaa32b9-e2f4-4d87-acfd-e0664a428f50]: {"event": "Finished scan profile recalculation [or
g=aa] [dur=0.21s]", "level": "info", "timestamp": "2024-08-07T08:08:23.498143"}
octopoes_api_worker-1  | [2024-08-07 08:08:23 +0000] [10] [ERROR] [service] Error running inference
octopoes_api_worker-1  | Traceback (most recent call last):
octopoes_api_worker-1  |   File "/app/octopoes/octopoes/core/service.py", line 224, in _run_inference
octopoes_api_worker-1  |     resulting_oois = BitRunner(bit_definition).run(source, parameters, config=config)
octopoes_api_worker-1  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
octopoes_api_worker-1  |   File "/app/octopoes/bits/runner.py", line 32, in run
octopoes_api_worker-1  |     return list(module.run(*args, **kwargs))
octopoes_api_worker-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
octopoes_api_worker-1  |   File "/app/octopoes/bits/domain_owner_verification/domain_owner_verification.py", line 19, in run
octopoes_api_worker-1  |     if DNSNSRecord.name_server_hostname.rstrip(".").upper() in INDICATORS:
octopoes_api_worker-1  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
octopoes_api_worker-1  |   File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 237, in __getattr__
octopoes_api_worker-1  |     raise AttributeError(item)
octopoes_api_worker-1  | AttributeError: name_server_hostname
```

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good.  Added a DNSNSRecord for mispo.es with one of the NS servers as mentioned above and this triggers the finding. 

image


### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by dekkers — APPROVED** [review]

**Pull request merged**